src/Core/PackageKitBackend.vala

```diff
@@ -474,10 +474,17 @@ public class AppCenterCore.PackageKitBackend : Backend, Object {
 
         Pk.Exit exit_status = Pk.Exit.UNKNOWN;
         string[] packages_ids = {};
-        foreach (var pk_package in package.change_information.updatable_ids) {
+        foreach (var pk_package in package.change_information.updatable_packages[this]) {
             packages_ids += pk_package;
         }
 
+        if (packages_ids.length == 0) {
+            job.result = Value (typeof (bool));
```

Comment:
I'm not sure if that will work, but something tells me that `job.result = true` will work, because I believe Vala will automatically translate native expressions to a `Value`.